### PR TITLE
Add trsv benchmark tools and sparselib format

### PR DESCRIPTION
**Pull request by upsj**
This PR adds tools to extract lower and upper triangular parts of a matrix (plus missing diagonal entries) for benchmarking purposes, as well as the sparselib Csr format.

### CODE REVIEW

File: benchmark/tools/make_lower_triangular.cpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#include <ginkgo/core/base/mtx_io.hpp>
+#include <iostream>
+#include <vector>
+
+int main()
+{
+    auto data = gko::read_raw<double, gko::int64>(std::cin);
```
Comment (yhmtsai) [inline benchmark/tools/make_lower_triangular.cpp:7]: need some usage description

File: benchmark/tools/make_upper_triangular.cpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    for (auto entry : data.nonzeros) {
+        if (entry.column >= entry.row) {
+            out.nonzeros.push_back(entry);
+        }
+        if (entry.column == entry.row) {
+            has_diag[entry.row] = true;
+        }
+    }
+    for (gko::int64 row = 0; row < data.size[0]; row++) {
+        if (!has_diag[row]) {
```
Comment (yhmtsai) [inline benchmark/tools/make_upper_triangular.cpp:20]: always have diagonal no matter if it is upper/lower triangular, right?

File: benchmark/tools/make_lower_triangular.cpp [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    gko::matrix_data<double, gko::int64> out(data.size);
```
Comment (yhmtsai) [inline benchmark/tools/make_lower_triangular.cpp:8]: I am not sure the complex one is needed or not

**Review by yhmtsai — DISMISSED** [review]
LGTM. needs more documentation

File: benchmark/tools/make_upper_triangular.cpp [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+            out.nonzeros.emplace_back(row, row, 1.0);
+        }
+    }
+    out.ensure_row_major_order();
+    gko::write_raw(std::cout, out, gko::layout_type::coordinate);
+}
```
Comment (yhmtsai) [inline benchmark/tools/make_upper_triangular.cpp:26]: I think the end of line should be `}\n` not just `}` by editorconfig.
just comes to my mind, not a strict format requirement.

**Review by pratikvn — APPROVED** [review]
LGTM!

**Review by tcojean — APPROVED** [review]
LGTM

**Comment by upsj** [comment]
rebase!

**Pull request merged**